packages/oscal-react-library/src/components/OSCALCatalogManageGroup.tsx

```diff
@@ -0,0 +1,1052 @@
+  const [addBelow, setAddBelow] = useState(false);
+  const [newGroupParent, setNewGroupParent] = useState<Group>();
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const groupsAndSubs: Array<Group> = baseGroups.map((x) => ({
+    groupTitle: x.title,
+    groupID: x.id,
+    projectUUID: data.projectUUID,
+    parentID: x.parent_id,
+    subGroups: [],
+    indent: 0,
+    others: [],
+  }));
+
+  const orderGroups: Array<Group> = [];
+  function reorderGroupsAndSetIndent() {
+    const parent_ids: Array<string> = [];
+    groupsAndSubs.forEach((elt) => {
+      if (!orderGroups.includes(elt)) {
+        if (elt.parentID === "") {
+          elt.indent = 0;
+          orderGroups.push(elt);
+          parent_ids.push(elt.groupID);
+        }
+      }
+    });
+
+    console.log(" order len ", orderGroups.length, " : all group len ", groupsAndSubs.length);
+    let depth = 0;
+    while (depth < 5) {
+      groupsAndSubs.forEach((elt) => {
+        if (!orderGroups.includes(elt)) {
+          if (parent_ids.includes(elt.parentID)) {
+            const index = orderGroups.findIndex((parent) => parent.groupID === elt.parentID);
+            if (index >= 0) {
+              elt.indent = orderGroups[index].indent + 10;
+              orderGroups[index].subGroups.push(elt);
+              if (index + 1 >= orderGroups.length) {
+                orderGroups.push(elt);
+                parent_ids.push(elt.groupID);
+              } else {
+                orderGroups.splice(index + 1, 0, elt);
+                parent_ids.push(elt.groupID);
+              }
+            }
+          }
+        }
+      });
+      depth = depth + 1;
+    }
+    console.log(" order len ", orderGroups.length, " : all group len ", groupsAndSubs.length);
+  }
+
+  reorderGroupsAndSetIndent();
+
+  orderGroups.forEach((elt) => {
+    setSiblings(elt.subGroups);
+  });
+
+  const orphans = orderGroups.filter((elt) => elt.parentID === "");
+  if (orphans.length > 0) {
+    let sibling = orphans[0];
+    for (let i = 1; i < orphans.length; i++) {
+      orphans[i].rightSibling = sibling;
+      sibling = orphans[i];
+    }
+  }
+
+  console.log("fin groups after parents :", groupsAndSubs);
+  console.log("ordered groups  after parents", orderGroups);
+  console.log("orphans", orphans);
+
+  function EditGroupTitle(ID: string, Name: string) {
+    const rootFile = "projects/catalog_" + data.projectUUID + "/oscal_data.json";
+    const request_json = {
+      oscal_file: rootFile,
+      id: ID,
+      title: Name,
+    };
+    function addNewGroupSuccess(response: any) {
+      console.log("successful addition of a new group", response);
+    }
+    function addNewGroupFail(e: any) {
+      console.log("Fail to create a new group", e.statusText);
+    }
+    fetchTransaction("/edit_group_title", request_json, addNewGroupSuccess, addNewGroupFail);
+  }
+  function DeleteGroup(ID: string) {
```

Comment:
To keep our function naming aligned I would just use camel casing here: `deleteGroup`.